packages/authentication/test/unit/strategy-adapter.test.ts

```diff
@@ -0,0 +1,66 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {ShimRequest} from '../..';
+import {StrategyAdapter} from '../..';
+import {ParsedRequest} from '@loopback/core';
+
+const passportStrategy = require('passport-strategy');
+
+describe('Strategy Adapter', () => {
+  const mockUser: User = {id: 'mock-user', role: 'mock-role'};
+  interface User {
+    id: string;
+    role: string;
+  }
+
+  describe('authenticate()', () => {
+
+    it('calls the authenticate method of the strategy', (done) => {
+      const strategy = new passportStrategy();
+      let calledFlag = false;
+      strategy.authenticate = function() {
+        calledFlag = true;
+      };
+      const adapter = new StrategyAdapter(strategy);
+      const request = <ParsedRequest> {};
+      // tslint:disable-next-line:no-floating-promises
+      adapter.authenticate(request);
+      expect(calledFlag).to.be.true();
+      done();
+    });
+
+    it('adds success handler to the strategy', (done) => {
+      const strategy = new MockStrategy();
+      const adapter = new StrategyAdapter(strategy);
+      const request = <ParsedRequest> {};
+      // tslint:disable-next-line:no-floating-promises
+      adapter.authenticate(request)
+      .then((user: User) => {
```

Comment:
This does not work correctly when `.authenticate` returns a rejected promise - the rejected promise will end up in the background and eventually got reported via Node.js unhandled promise reporter. This is one of the reasons why `tslint` complained about a floating promise. Hiding the tslint error is not a proper fix ;-)

In general, don't mix `done` callback-style and promise-style code in tests. Mocha supports promises OOTB. We should use async/await to get even cleaner code.

```ts
it('adds success handler to the strategy', async () => {
  // ...
  const user = await adapter.authenticate(request);
  expect(user)//...
});
```